util_linux/fsck_minix.rs

```diff
@@ -619,50 +616,45 @@ unsafe extern "C" fn write_superblock() {
 
 unsafe extern "C" fn write_tables() {
   write_superblock();
-  if ((*ptr_to_globals).u.Super.s_imap_blocks as libc::c_uint)
-    .wrapping_mul(BLOCK_SIZE as libc::c_int as libc::c_uint) as libc::c_long
-    != write(
+
+  let imap_size = (*ptr_to_globals).u.Super.s_imap_blocks as isize * BLOCK_SIZE as isize;
```

Comment:
I think it ends up a little simpler in this case compared to:

```rust
let imap_size = ((*ptr_to_globals).u.Super.s_imap_blocks as u32 * BLOCK_SIZE) as isize;
```
For that whole block of code (including the `if` after) you always need to cast `imap_size` or it's components 3 times, it would just be a case of moving the casts.

Unless you can see a simpler transformation?

Reply:
Oh I see... yeah I don't see a cleaner alternative atm